### PR TITLE
Added Archived column to FileInfo

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Added an `archived` column to FileInfo table and corresponding field in the model for cloud freemium.

#### Ticket Link
https://community-daily.mattermost.com/boards/workspace/qgsck6cts3fwpqwyjiupjm5cde/b4ctmm917q3rzdfi451qtuxjzcw/vch9s93uimjrfik7fisjczzpfkc/cpwowhu7odbd5jdeox1nwjs94dr

#### Release Note
```release-note
MySQL self-hosted customers may notice the migration taking longer than usual when having a large number of rows in FileInfo table.
```

### CODE REVIEW

File: db/migrations/postgres/000084_fileinfo_add_archived_column.down.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE fileinfo DROP COLUMND IF EXISTS archived;
```
Comment (ashishbhate) [inline db/migrations/postgres/000084_fileinfo_add_archived_column.down.sql:1]: Typo: `COLUMND` -> `COLUMN`

Comment (harshilsharma63) [reply db/migrations/postgres/000084_fileinfo_add_archived_column.down.sql:1]: that's a bit too quick :P

Comment (agnivade) [reply db/migrations/postgres/000084_fileinfo_add_archived_column.down.sql]: Sending it too fast :)

Comment (harshilsharma63) [reply db/migrations/postgres/000084_fileinfo_add_archived_column.down.sql]: Done.

**Review by vish9812 — APPROVED** [review]

**Review by isacikgoz — APPROVED** [review]

**Review by ashishbhate — APPROVED** [review]
LGTM!


Since we're adding a `NOT NULL` column with a default, I think this will need a full table lock. Our community FileInfo table might be big enough for this to be a problem. We might have to give the SRE team a head's up about this PR and let them merge it when works best for them. 

@agnivade could you confirm if the above is correct?

**Comment by harshilsharma63** [comment]
Query performance -

Postgres -

```
mattermostdb=# select count(*) from fileinfo;
 count
--------
 753216
(1 row)

Time: 24.867 ms
mattermostdb=# ALTER TABLE fileinfo ADD COLUMN IF NOT EXISTS archived boolean NOT NULL DEFAULT false;
ALTER TABLE
Time: 1.388 ms
```

MySQL-

```
mysql> select count(*) from fileinfo;
+----------+
| count(*) |
+----------+
|   737280 |
+----------+
1 row in set (0.31 sec)

mysql> ALTER TABLE FileInfo ADD COLUMN Archived boolean NOT NULL DEFAULT false;
Query OK, 737280 rows affected (19.22 sec)
Records: 737280  Duplicates: 0  Warnings: 0

mysql> show profiles;
+----------+-------------+-------------------------------------------------------------------------+
| Query_ID | Duration    | Query                                                                   |
+----------+-------------+-------------------------------------------------------------------------+
|        1 |  0.30536400 | select count(*) from fileinfo                                           |
|        2 | 19.22379400 | ALTER TABLE FileInfo ADD COLUMN Archived boolean NOT NULL DEFAULT false |
+----------+-------------+-------------------------------------------------------------------------+
```

**Comment by harshilsharma63** [comment]
> Nice work @harshilsharma63 ! Any chance you have the DB specs for MySQL and Postgres? It would be helpful to mention that as well.

What's a DB spec? I'm doing this for the first time.

**Comment by agnivade** [comment]
The CPU/Mem details of the DB instance.

**Comment by harshilsharma63** [comment]
@agnivade here they re-

CPU - Intel i7 6 cores @ 2.6 GHz
Memory - 16 GB

**Pull request merged**